resdk/resolwe.py

```diff
@@ -293,16 +293,23 @@ def run(self, slug=None, input={}, descriptor=None,  # pylint: disable=redefined
         else:
             raise ValueError("Unexpected behaviour at get process with slug {}".format(slug))
 
-        # Upload files in basic:file fields
+        # Pre-process inputs
         try:
             for schema, fields in iterate_fields(input, process['input_schema']):
                 field_name = schema['name']
                 field_type = schema['type']
                 field_value = fields[field_name]
 
+                # Wrapp `list:` fields into list if they are not already
```

Comment:
Now that I have slept on this, I think we should raise an error. We are building an SDK and exceptions are fine in libraries. I agree that for the user working in Python interpreter this is not convenient, but imagine a software package like Orange using our Python SDK. Exceptions should get raised for wrong inputs. Otherwise our SDK will get as messy as JavaScript :) Where you never know what evaluates to True and what not. What we want here is a ValidationError that is raised for wrong inputs. The validation can happen locally or on the server. If locally, it would work faster, but we will have to keep it in sync with the validation on the server. If on the server, we would have to capture the response, parse and reformat the exception.